ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -63,9 +71,10 @@ data IsLedgerPeer = IsLedgerPeer | IsNotLedgerPeer
 
 newtype NumberOfPeers = NumberOfPeers Word16 deriving Show
 
-newtype LedgerPeersConsensusInterface m = LedgerPeersConsensusInterface {
-      lpGetPeers :: SlotNo -> STM m (Maybe [(PoolStake, NonEmpty RelayAccessPoint)])
-    }
+data LedgerPeers = LedgerPeers LedgerStateJudgement
+                               [(PoolStake, NonEmpty RelayAccessPoint)]
+                 | BeforeSlot
```

Comment:
Could you add some haddocks, `BeforeSlot` makes sense only if one knows the context.